### DISCUSSION TITLE
Defining project property "TargetFrameworks" breaks Avalonia.BuildService [Avalonia 11.0.0-preview8]

### DISCUSSION THREAD

**Discussion by BinToss** [AvaloniaUI/Avalonia#11567] [General]
```diff
- C:\Users\NoahR\.nuget\packages\avalonia.buildservices\0.0.12\buildTransitive\Avalonia.BuildServices.targets(4,16):
- error MSB4022: The result "" of evaluating the value "$(AvaloniaBuildServicesTasksLocation)" of the "AssemblyFile" attribute in element <UsingTask> is not valid. [C:\Repos\HaloSPV3\HXE\src\HXE.csproj]
```
If this property is removed from my project file-removing multi-targeting-the error does not occur and the project proceeds to build successfully. [This property is supposed to be ignored when `TargetFramework` is defined.](https://learn.microsoft.com/en-us/dotnet/core/project-sdk/msbuild-props#targetframeworks).

**Comment by danwalmsley** [comment]
@BinToss iv tried to reproduce this but am unable, can you provide what value you set in `<TargetFrameworks>`? Or better still a zip with a minimal project to reproduce.

**Reply by rabbitism** [reply]
My experience:
1. create an empty Avalonia app (net6.0 by default)
2. build and run
3. Change `<TargetFramework>net6.0</TargetFramework>` to `<TargetFrameworks>net6.0; net7.0</TargetFrameworks>`
4. build and run
5. Change back to `<TargetFramework>net6.0</TargetFramework>` 

then the project cannot load into the solution anymore, with error in this discussion.

**Reply by BinToss** [reply]
Exactly what @rabbitism said.

**Reply by danwalmsley** [reply]
thansk I reproduced it.

**Reply by danwalmsley** [reply]
https://github.com/AvaloniaUI/Avalonia/pull/11585

**Reply by BinToss** [reply]
<details>
Issue still occurs in 11.0.0-rc1.1
I discovered `TargetFrameworks` value is unrelated.
Build any Avalonia-dependent project with `--framework net6.0`. Then build it again with `--framework net7.0`. Or vice versa.
Attempting to build again with the framework targeted in the successful build will still result in the error.
Deleting the `obj` folder allows builds to succeed until a different framework is targeted.

```diff
MSBuild version 17.6.1+8ffc3fe3d for .NET
Build started 2023-06-06 3:14:01 AM.
     1>Project "C:\Repos\HaloSPV3\HXE\src\HXE.sln" on node 1 (Clean target(s)).
     1>ValidateSolutionConfiguration:
         Building solution configuration "Debug|Any CPU".
     1>Project "C:\Repos\HaloSPV3\HXE\src\HXE.sln" (1) is building "C:\Repos\HaloSPV3\HXE\src\HXE.csproj" (2) on node 1 (Clean target(s)).
-    2>C:\Users\NoahR\.nuget\packages\avalonia\11.0.0-rc1.1\buildTransitive\AvaloniaBuildTasks.targets(25,14):
-      error MSB4022: The result "" of evaluating the value "$(AvaloniaBuildTasksLocation)" of the "AssemblyFile" attribute in element <UsingTask> is not valid. [C:\Repos\HaloSPV3\HXE\src\HXE.csproj]
     2>Done Building Project "C:\Repos\HaloSPV3\HXE\src\HXE.csproj" (Clean target(s)) -- FAILED.
     1>Done Building Project "C:\Repos\HaloSPV3\HXE\src\HXE.sln" (Clean target(s)) -- FAILED.
```
</details>
I don't want to create an issue until it's properly confirmed to be a bug.

----

The issue does *not* occur in a new project created via `dotnet new avalonia.app -n avalonia-frameworks-issue -av 11.0.0-rc1.1` with `TargetFramework` changed to `TargetFrameworks` with `;net7.0` added. I also had to add `RuntimeIdentifiers` to check specific runtimes. 

EDIT: now, after deleting `obj` again, I'm unable to reproduce the issue in my "HXE" project with any framework and runtime combination.

**Comment by BinToss** [comment]
If anyone finds this again, beware of .NET's lack of warning when you define both `TargetFramework `and `TargetFrameworks`. If you have the latter defined, the former should only be defined by the CLI's `--framework <FRAMEWORK>` argument or programmatically.
